src/main/java/htsjdk/variant/variantcontext/GenotypeNumLikelihoodsCache.java

```diff
@@ -0,0 +1,43 @@
+package htsjdk.variant.variantcontext;
+
+import java.util.HashMap;
+
+/**
+ * Hybrid dynamic cache for genotype likelihood counts
+ */
+
+public class GenotypeNumLikelihoodsCache {
+
+    private final static int N_ALLELES = 5;
+    private final static int PLOIDY = 10;
+
+    private final int[][] staticCache = new int[5][10];
+    private final HashMap<Integer, HashMap<Integer, Integer>> dynamicCache = new HashMap<>();
+
+    public void put(final int numAlleles, final int ploidy, final int numLikelihoods) {
+        if(numAlleles < N_ALLELES && ploidy < PLOIDY) {
+            staticCache[numAlleles][ploidy] = numLikelihoods;
+        }
+        else{
+            final HashMap<Integer, Integer> cache = dynamicCache.containsKey(ploidy) ? dynamicCache.get(ploidy) : new HashMap<>();
```

Comment:
This is kind of nitpicky, but since we're saying this is important to performance ( which hopefully it's much less important now that we have a faster implementation), but we might want to avoid redundant hash look ups by using get and saving / checking the result rather than the nicer contains followed by a get.  I might skip that though since I don't think it will be a bottleneck at all...

Reply:
This is an easy enough change.  Although switching to a compound will probably remove this